core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,196 @@
+    }
+
+    pub async fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let pool = self.pool.clone();
+        loop {
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, da_dispatcher is shutting down");
+                break;
+            }
+
+            tokio::join!(
+                async {
+                    if let Err(err) = self.dispatch(&pool).await {
+                        tracing::error!("dispatch error {err:?}");
+                    }
+                },
+                async {
+                    if let Err(err) = self.poll_for_inclusion(&pool).await {
+                        tracing::error!("poll_for_inclusion error {err:?}");
+                    }
+                }
+            );
+
+            tokio::time::sleep(self.config.polling_interval()).await;
+        }
+        Ok(())
+    }
+
+    /// Dispatches the blobs to the data availability layer, and saves the blob_id in the database.
+    async fn dispatch(&self, pool: &ConnectionPool<Core>) -> anyhow::Result<()> {
+        let mut conn = pool.connection_tagged("da_dispatcher").await?;
+        let batches = conn
+            .data_availability_dal()
+            .get_ready_for_da_dispatch_l1_batches(self.config.max_rows_to_dispatch() as usize)
+            .await?;
+        drop(conn);
+
+        for batch in batches {
+            let dispatch_latency = METRICS.blob_dispatch_latency.start();
+            let dispatch_response = retry(self.config.max_retries(), batch.l1_batch_number, || {
+                self.client
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .with_context(|| {
+                format!(
+                    "failed to dispatch a blob with batch_number: {}, pubdata_len: {}",
+                    batch.l1_batch_number,
+                    batch.pubdata.len()
+                )
+            })?;
+            let dispatch_latency_duration = dispatch_latency.observe();
+
+            let sent_at =
+                NaiveDateTime::from_timestamp_millis(Utc::now().timestamp_millis()).unwrap();
+
+            let mut conn = pool.connection_tagged("da_dispatcher").await?;
+            conn.data_availability_dal()
+                .insert_l1_batch_da(
+                    batch.l1_batch_number,
+                    dispatch_response.blob_id.as_str(),
+                    sent_at,
+                )
+                .await?;
+            drop(conn);
+
+            METRICS
+                .last_dispatched_l1_batch
+                .set(batch.l1_batch_number.0 as usize);
+            METRICS.blob_size.observe(batch.pubdata.len());
+            tracing::info!(
+                "Dispatched a DA for batch_number: {}, pubdata_size: {}, dispatch_latency ms: {}",
+                batch.l1_batch_number,
+                batch.pubdata.len(),
+                dispatch_latency_duration.as_millis()
```

Comment:
Nit: You can output `{dispatch_latency_duration:?}` in the log directly; it's human-readable.